### PR TITLE
techdebt: [CDS-14583]: Exception Handling with Exception Sanitizer for other swimlane

### PR DESCRIPTION
**Pull request by GoelPratyush99**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/32129)

### CODE REVIEW

**Comment by GoelPratyush99** [comment]
Recency: trigger recency

**Review by abosii-harness — CHANGES_REQUESTED** [review]
> Reviewed 24 of 32 files at r1, 78 of 80 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32129)* status: Some of the checks are too old. 1 change requests. 0 of 1 approvals obtained. 

___
*[260-delegate/src/main/java/software/wings/delegatetasks/aws/AwsS3Task.java, line 64 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0AhPYWAbNiX1FkfUpQ:-N0AhPYWAbNiX1FkfUpR:b-l00uhe) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/260-delegate/src/main/java/software/wings/delegatetasks/aws/AwsS3Task.java#L64)):*
> ```Java
>       }
>     } catch (WingsException exception) {
>       throw(WingsException) ExceptionMessageSanitizer.sanitizeException(exception);
> ```

nit: space after throw
___
*[260-delegate/src/test/java/software/wings/delegatetasks/TerraformProvisionTaskTest.java, line 183 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0Ah_PlBlzDf5wyhkat:-N0Ah_PlBlzDf5wyhkau:bhwdsng) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/260-delegate/src/test/java/software/wings/delegatetasks/TerraformProvisionTaskTest.java#L183)):*
> ```Java
> 
>     sourceRepoEncryptionDetails = new ArrayList<>();
>     sourceRepoEncryptionDetails.add(EncryptedDataDetail.builder().fieldName("fieldName").build());
> ```

Can we add this case in Sanitizer itself so we would handle if fieldName is empty rather than updating unit test
___
*[400-rest/src/main/java/software/wings/delegatetasks/aws/AwsCFTask.java, line 81 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0AhooBB8rUP2D1A47s:-N0AhooC3fzial_qvxfd:b-l00uhe) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/400-rest/src/main/java/software/wings/delegatetasks/aws/AwsCFTask.java#L81)):*
> ```Java
>       }
>     } catch (WingsException exception) {
>       throw(WingsException) ExceptionMessageSanitizer.sanitizeException(exception);
> ```

nit: space after throw
___
*[400-rest/src/main/java/software/wings/delegatetasks/aws/AwsCodeDeployTask.java, line 117 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0Ahr-R40hLRYaYpN0O:-N0Ahr-R40hLRYaYpN0P:b-l00uhe) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/400-rest/src/main/java/software/wings/delegatetasks/aws/AwsCodeDeployTask.java#L117)):*
> ```Java
>       }
>     } catch (WingsException exception) {
>       throw(WingsException) ExceptionMessageSanitizer.sanitizeException(exception);
> ```

nit: space after throw
___
*[400-rest/src/main/java/software/wings/delegatetasks/aws/AwsEc2Task.java, line 116 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0Ahsby-xa_alFI8r4L:-N0Ahsby-xa_alFI8r4M:b-l00uhe) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/400-rest/src/main/java/software/wings/delegatetasks/aws/AwsEc2Task.java#L116)):*
> ```Java
>       }
>     } catch (WingsException exception) {
>       throw(WingsException) ExceptionMessageSanitizer.sanitizeException(exception);
> ```

nit: space after throw
___
*[400-rest/src/main/java/software/wings/delegatetasks/aws/AwsEcsTask.java, line 83 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0AhvX89GsXrnwWkGqD:-N0AhvX89GsXrnwWkGqE:b-l00uhe) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/400-rest/src/main/java/software/wings/delegatetasks/aws/AwsEcsTask.java#L83)):*
> ```Java
>       }
>     } catch (WingsException exception) {
>       throw(WingsException) ExceptionMessageSanitizer.sanitizeException(exception);
> ```

nit: space after throw
___
*[400-rest/src/main/java/software/wings/delegatetasks/aws/AwsElbTask.java, line 109 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0AhwH236PgY0NQdU-h:-N0AhwH236PgY0NQdU-i:b-l00uhe) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/400-rest/src/main/java/software/wings/delegatetasks/aws/AwsElbTask.java#L109)):*
> ```Java
>       }
>     } catch (WingsException exception) {
>       throw(WingsException) ExceptionMessageSanitizer.sanitizeException(exception);
> ```

nit: space after throw
___
*[400-rest/src/main/java/software/wings/delegatetasks/azure/taskhandler/AzureVMSSTaskHandler.java, line 187 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0Ai0ni6DCyLiTZ-KlW:-N0Ai0ni6DCyLiTZ-KlX:b-l00uhe) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/400-rest/src/main/java/software/wings/delegatetasks/azure/taskhandler/AzureVMSSTaskHandler.java#L187)):*
> ```Java
>       throw new InvalidRequestException(message, e);
>     } catch (WingsException e) {
>       throw(WingsException) ExceptionMessageSanitizer.sanitizeException(e);
> ```

nit: space after throw
___
*[400-rest/src/main/java/software/wings/service/impl/aws/delegate/AwsLambdaHelperServiceDelegateImpl.java, line 628 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0AiMBu44PK-YBr-1EH:-N0AiMBu44PK-YBr-1EI:b-l00uhe) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/400-rest/src/main/java/software/wings/service/impl/aws/delegate/AwsLambdaHelperServiceDelegateImpl.java#L628)):*
> ```Java
>           "Timeout", ExceptionMessageSanitizer.sanitizeException(e), WingsException.SRE);
>     } catch (WingsException e) {
>       throw(WingsException) ExceptionMessageSanitizer.sanitizeException(e);
> ```

nit: space after throw
___
*[400-rest/src/test/java/software/wings/delegatetasks/TerragruntProvisionTaskHelperTest.java, line 154 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0AiPoGE0cINdWhNafC:-N0AiPoGE0cINdWhNafD:bwx4k4p) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/400-rest/src/test/java/software/wings/delegatetasks/TerragruntProvisionTaskHelperTest.java#L154)):*
> ```Java
>             .gitFileConfig(gitFileConfig)
>             .gitConfig(GitConfig.builder().build())
>             .encryptedDataDetails(singletonList(EncryptedDataDetail.builder().fieldName("fieldName").build()))
> ```

same here, let's have a handler in sanitizer
___
*[400-rest/src/test/java/software/wings/delegatetasks/TerragruntProvisionTaskTest.java, line 146 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0AiTQY8n_fGgSG61jY:-N0AiTQY8n_fGgSG61jZ:b-ffynuv) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/400-rest/src/test/java/software/wings/delegatetasks/TerragruntProvisionTaskTest.java#L146)):*
> ```Java
> 
>     sourceRepoEncryptionDetails = new ArrayList<>();
>     sourceRepoEncryptionDetails.add(EncryptedDataDetail.builder().fieldName("fieldName").build());
> ```

let's have a handling of missing in sanitizer itself and revert this
___
*[970-api-services-beans/BUILD.bazel, line 15 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0AinLC-bUgifO2l5V2:-N0AinLC-bUgifO2l5V3:bs24sun) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/970-api-services-beans/BUILD.bazel#L15)):*
> ```Starlark
>         "//:lombok",
>         "//970-api-services-beans/src/main/proto:970_api_services_beans_java_proto",
>         "//970-ng-commons:module",
> ```

why this is required?
___
*[970-api-services-beans/src/main/java/io/harness/serializer/ApiServiceBeansRegistrars.java, line 27 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0Air31-TfaQ6hl25-w:-N0Air31-TfaQ6hl25-x:b-crem8h) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/970-api-services-beans/src/main/java/io/harness/serializer/ApiServiceBeansRegistrars.java#L27)):*
> ```Java
>           .add(ApiServiceBeansKryoRegister.class)
>           .add(ApiServiceBeansProtoKryoRegistrar.class)
>           .add(RecasterKryoRegistrar.class)
> ```

Why this is required?

**Review by namantalaycha10 — APPROVED** [review]

**Review by namantalaycha10 — DISMISSED** [review]
> Reviewed 24 of 32 files at r1, 78 of 80 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32129)* status: Some of the checks are too old. 1 change requests.

**Comment by arvind-choudhary-h** [comment]
*[930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/KubernetesTestConnectionDelegateTask.java, line 39 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0FFMcf59uduJuPDMXK:-N0FFMcg4RrYPyguRdS6:bowibfb) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/KubernetesTestConnectionDelegateTask.java#L39)):*
> ```Java
>       BooleanSupplier preExecute) {
>     super(delegateTaskPackage, logStreamingTaskClient, consumer, preExecute);
>     SecretSanitizerThreadLocal.addAll(delegateTaskPackage.getSecrets());
> ```

@goelpratyush99 / @namantalaycha10  - shouldn't we add this line in public AbstractDelegateRunnableTask() constructor? What are pros and cons? I feel this is duplicate.
cc - @abosii-harness

**Comment by arvind-choudhary-h** [comment]
*[930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/KubernetesTestConnectionDelegateTask.java, line 39 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0FFMcf59uduJuPDMXK:-N0FFiTGCsKllH35KAKc:buf0x1e) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/KubernetesTestConnectionDelegateTask.java#L39)):*
<details><summary><i>Previously, arvind-choudhary-h (Arvind Choudhary) wrote…</i></summary><blockquote>

@goelpratyush99 / @namantalaycha10  - shouldn't we add this line in public AbstractDelegateRunnableTask() constructor? What are pros and cons? I feel this is duplicate.
cc - @abosii-harness  
</blockquote></details>

And not just here but all for the classes implements DelegateRunnableTask

**Comment by arvind-choudhary-h** [comment]
*[930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/KubernetesTestConnectionDelegateTask.java, line 39 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0FFMcf59uduJuPDMXK:-N0FFo3T1Edu_BWV20Hd:bi4gzbf) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/KubernetesTestConnectionDelegateTask.java#L39)):*
<details><summary><i>Previously, arvind-choudhary-h (Arvind Choudhary) wrote…</i></summary><blockquote>

And not just here but all for the classes implements DelegateRunnableTask
</blockquote></details>

Sorry - extends AbstractDelegateRunnableTask

**Review by arvind-choudhary-h — COMMENTED** [review]
> Reviewed 24 of 32 files at r1, 77 of 80 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32129)* status: Some of the checks are too old. 1 change requests.

**Review by GoelPratyush99 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32129)* status: 105 file(s) to review. 1 change requests. 14. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/delegatetasks/aws/AwsCFTask.java` line 81 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0AhooBB8rUP2D1A47s:-N0VTTjM3TzoVCn6dv9W:b-1q26yj) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/400-rest/src/main/java/software/wings/delegatetasks/aws/AwsCFTask.java#L81)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

nit: space after throw
</blockquote></details>

Its because of clang-format
___
*[`400-rest/src/main/java/software/wings/delegatetasks/aws/AwsCodeDeployTask.java` line 117 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0Ahr-R40hLRYaYpN0O:-N0VTb-e7CjyMr8UMpzo:b-1q26yj) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/400-rest/src/main/java/software/wings/delegatetasks/aws/AwsCodeDeployTask.java#L117)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

nit: space after throw
</blockquote></details>

Its because of clang-format
___
*[`400-rest/src/main/java/software/wings/delegatetasks/aws/AwsEc2Task.java` line 116 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0Ahsby-xa_alFI8r4L:-N0VTdek9IPxfmHmDqz7:b-1q26yj) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/400-rest/src/main/java/software/wings/delegatetasks/aws/AwsEc2Task.java#L116)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

nit: space after throw
</blockquote></details>

Its because of clang-format
___
*[`400-rest/src/main/java/software/wings/delegatetasks/aws/AwsEcsTask.java` line 83 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0AhvX89GsXrnwWkGqD:-N0VTew9-EaZCHCQ47bC:bsb9lpq) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/400-rest/src/main/java/software/wings/delegatetasks/aws/AwsEcsTask.java#L83)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

nit: space after throw
</blockquote></details>

Its because of clang-format. Fixed it
___
*[`400-rest/src/main/java/software/wings/delegatetasks/aws/AwsElbTask.java` line 109 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0AhwH236PgY0NQdU-h:-N0VTii2-g0sPmNd-MTH:bsb9lpq) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/400-rest/src/main/java/software/wings/delegatetasks/aws/AwsElbTask.java#L109)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

nit: space after throw
</blockquote></details>

Its because of clang-format. Fixed it
___
*[`400-rest/src/main/java/software/wings/service/impl/aws/delegate/AwsLambdaHelperServiceDelegateImpl.java` line 628 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0AiMBu44PK-YBr-1EH:-N0VTmNt9lCJs2Dphzh2:bsb9lpq) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/400-rest/src/main/java/software/wings/service/impl/aws/delegate/AwsLambdaHelperServiceDelegateImpl.java#L628)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

nit: space after throw
</blockquote></details>

Its because of clang-format. Fixed it
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/KubernetesTestConnectionDelegateTask.java` line 39 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0FFMcf59uduJuPDMXK:-N0VToePDdQMVEjmnl-9:b4gpoc4) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/KubernetesTestConnectionDelegateTask.java#L39)):*
<details><summary><i>Previously, arvind-choudhary-h (Arvind Choudhary) wrote…</i></summary><blockquote>

Sorry - extends AbstractDelegateRunnableTask
</blockquote></details>

At that time I thought I could not edit any file that is not owned by CDP. I have fixed it, now the code snippet will be called from inside AbstractDelegateRunnableTask constructor.



_Code snippet:_
```Java
SecretSanitizerThreadLocal.addAll(delegateTaskPackage.getSecrets());
```
___
*[`970-api-services-beans/BUILD.bazel` line 15 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0AinLC-bUgifO2l5V2:-N0VUIkZ1coQlMjl4KLl:b21z77t) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/970-api-services-beans/BUILD.bazel#L15)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

why this is required?
</blockquote></details>

As ExceptionMessageSanitizer class is in 970-api-service-beans. It was throwing some kryo error for NGCommonsKryoRegistrar and RecasterKryoRegistrar. Hence to fix it we had to add 970-ng-commons:module.
___
*[`970-api-services-beans/src/main/java/io/harness/serializer/ApiServiceBeansRegistrars.java` line 27 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0Air31-TfaQ6hl25-w:-N0VWUbc-jdnNpCcCcJp:bsbc4uv) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/970-api-services-beans/src/main/java/io/harness/serializer/ApiServiceBeansRegistrars.java#L27)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

Why this is required?
</blockquote></details>

As ExceptionMessageSanitizer class is in 970-api-service-beans. It was throwing some kryo error for NGCommonsKryoRegistrar and RecasterKryoRegistrar. Hence to fix it we had to add these after discussing with @tchaurasiya 
___
*[`260-delegate/src/main/java/software/wings/delegatetasks/aws/AwsS3Task.java` line 64 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0AhPYWAbNiX1FkfUpQ:-N0VWfFD0eHwPWQcQlXw:b2wxoid) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/260-delegate/src/main/java/software/wings/delegatetasks/aws/AwsS3Task.java#L64)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

nit: space after throw
</blockquote></details>

Error from clang-format. fixed it
___
*[`260-delegate/src/test/java/software/wings/delegatetasks/TerraformProvisionTaskTest.java` line 183 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0Ah_PlBlzDf5wyhkat:-N0VWkCs2ZPIhpbvyC3-:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/260-delegate/src/test/java/software/wings/delegatetasks/TerraformProvisionTaskTest.java#L183)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

Can we add this case in Sanitizer itself so we would handle if fieldName is empty rather than updating unit test
</blockquote></details>

done
___
*[`400-rest/src/main/java/software/wings/delegatetasks/azure/taskhandler/AzureVMSSTaskHandler.java` line 187 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0Ai0ni6DCyLiTZ-KlW:-N0VWn-jCMwcgivwZFKf:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/400-rest/src/main/java/software/wings/delegatetasks/azure/taskhandler/AzureVMSSTaskHandler.java#L187)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

nit: space after throw
</blockquote></details>

done
___
*[`400-rest/src/test/java/software/wings/delegatetasks/TerragruntProvisionTaskHelperTest.java` line 154 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0AiPoGE0cINdWhNafC:-N0VWoY_7u-P17YnWlGL:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/400-rest/src/test/java/software/wings/delegatetasks/TerragruntProvisionTaskHelperTest.java#L154)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

same here, let's have a handler in sanitizer
</blockquote></details>

done
___
*[`400-rest/src/test/java/software/wings/delegatetasks/TerragruntProvisionTaskTest.java` line 146 at r3](https://reviewable.io/reviews/harness/harness-core/32129#-N0AiTQY8n_fGgSG61jY:-N0VWq2e8R_sY7JMkWVH:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/fe92ed198c89bcea329f15a1fbbda4a12f9888a6/400-rest/src/test/java/software/wings/delegatetasks/TerragruntProvisionTaskTest.java#L146)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

let's have a handling of missing in sanitizer itself and revert this
</blockquote></details>

done

File: 930-delegate-tasks/src/main/java/software/wings/delegatetasks/TerragruntProvisionTask.java [hunk 1]
```diff
@@ -433,10 +439,12 @@ private TerragruntExecutionData run(TerragruntProvisionParameters parameters) {
       return terragruntExecutionDataBuilder.build();
 
     } catch (WingsException ex) {
-      return logErrorAndGetFailureResponse(wrapUpLogCallBack, ex, ExceptionUtils.getMessage(ex));
-    } catch (IOException ex) {
       return logErrorAndGetFailureResponse(
-          wrapUpLogCallBack, ex, format("IO Failure occurred while performing Terragrunt Task: %s", ex.getMessage()));
+          wrapUpLogCallBack, ex, ExceptionUtils.getMessage(ExceptionMessageSanitizer.sanitizeException(ex)));
```
Comment (namantalaycha10) [inline 930-delegate-tasks/src/main/java/software/wings/delegatetasks/TerragruntProvisionTask.java:443]: @GoelPratyush99 can you also sanitize this ex before passing in logErrorAndGetFailureResponse

**Review by namantalaycha10 — CHANGES_REQUESTED** [review]

**Review by GoelPratyush99 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32129)* status: 195 file(s) to review. 2 change requests. 15. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/software/wings/delegatetasks/TerragruntProvisionTask.java` line 443 at r5](https://reviewable.io/reviews/harness/harness-core/32129#-N2W07i-Xs-SLN7nIDUA-r5-443:-N3NTs0k6KKMJppGFViT:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/1e3ab5b8e42d23e6f5351d36d34330e2f2474c69/930-delegate-tasks/src/main/java/software/wings/delegatetasks/TerragruntProvisionTask.java#L443)):*
<details><summary><i>Previously, namantalaycha10 (Naman Talaycha) wrote…</i></summary><blockquote>

@GoelPratyush99 can you also sanitize this ex before passing in logErrorAndGetFailureResponse
</blockquote></details>

Done

**Review by abosii-harness — APPROVED** [review]
> Reviewed 18 of 105 files at r4, 176 of 177 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32129)* status: Some of the checks are too old. 1 change requests. 3. unresolved discussions.

**Review by namantalaycha10 — COMMENTED** [review]
> Reviewed 18 of 105 files at r4, 176 of 177 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32129)* status: Some of the checks are too old. 1 change requests. 3. unresolved discussions.

**Review by namantalaycha10 — APPROVED** [review]

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 1 of 105 files at r4, 89 of 177 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32129)* status: Some of the checks are too old. 3. unresolved discussions.

**Review by raghuAtWings — APPROVED** [review]
> Reviewed 1 of 105 files at r4, 89 of 177 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32129)* status: Some of the checks are too old. 3. unresolved discussions.

**Pull request merged**